aptos-move/aptos-transaction-simulation-session/src/session.rs

```diff
@@ -414,7 +428,107 @@ impl Session {
         res
     }
 
-    // TODO: view resource
+    /// Views a Move resource.
+    pub fn view_resource(
+        &mut self,
+        account_addr: AccountAddress,
+        resource_tag: &StructTag,
+    ) -> Result<Option<serde_json::Value>> {
+        let state_key = StateKey::resource(&account_addr, resource_tag)?;
+
+        let json_val = match self.state_store.get_state_value_bytes(&state_key)? {
+            Some(bytes) => {
+                let annotator = AptosValueAnnotator::new(&self.state_store);
+                let annotated =
+                    AnnotatedMoveValue::Struct(annotator.view_resource(resource_tag, &bytes)?);
+                Some(aptos_api_types::MoveValue::try_from(annotated)?.json()?)
+            },
+            None => None,
+        };
+
+        let summary = Summary::ViewResource {
+            resource_type: resource_tag.to_canonical_string(),
+            resource_value: json_val.clone(),
+        };
+
+        let summary_path = self
+            .path
+            .join(format!(
+                "[{}] view resource 0x{}::0x{}::{}::{}",
+                self.config.ops,
+                format_address(&account_addr),
+                format_address(&resource_tag.address),
+                resource_tag.module,
+                resource_tag.name,
+            ))
+            .join("summary.json");
+        std::fs::create_dir_all(summary_path.parent().unwrap())?;
+        std::fs::write(summary_path, serde_json::to_string_pretty(&summary)?)?;
+
+        self.config.ops += 1;
+        self.config.save_to_file(&self.path.join("config.json"))?;
+
+        Ok(json_val)
+    }
+
+    pub fn view_resource_group(
+        &mut self,
+        account_addr: AccountAddress,
+        resource_group_tag: &StructTag,
+        derived_object_address: Option<AccountAddress>,
+    ) -> Result<Option<serde_json::Value>> {
+        let account_addr = match derived_object_address {
+            Some(addr) => create_derived_object_address(account_addr, addr),
+            None => account_addr,
+        };
+
+        let state_key = StateKey::resource_group(&account_addr, resource_group_tag);
+
+        let json_val = match self.state_store.get_state_value_bytes(&state_key)? {
+            Some(bytes) => {
+                let group: BTreeMap<StructTag, Vec<u8>> = bcs::from_bytes(&bytes)?;
```

Comment:
nit: I believe we have `ObjectGroup` type defined in `aptos-types` which you can re-use (wrapper around btree map)